### PR TITLE
TASK-2951, TASK-2952: Updated letters endpoint and interop service

### PR DESCRIPTION
**Pull request by carolinegeorge-gov**
### Description

- Renamed `letter-service` to `interop-service`
- Added getAllLetterTypes mocks and endpoint

### Related Azure Boards Work Items

[AB#2951](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/2951)
[AB#2952](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/2952)

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### Test Instructions
1. Add the `interop-service` to the page, if not already there
2. Call `getAllLetterTypes`

### CODE REVIEW

File: frontend/app/services/interop-service.server.ts [hunk 1]
```diff
@@ -18,12 +18,22 @@ const letterSchema = z.array(
...
+
+const listOfLetterTypeCodeSchema = z.array(letterTypeCodeSchema);
+export type LetterTypeCodeList = z.infer<typeof listOfLetterTypeCodeSchema>;
+
 /**
- * Return a singleton instance (by means of memomization) of the letter service.
+ * Return a singleton instance (by means of memomization) of the interop service.
  */
-export const getLettersService = moize(createLettersService, { onCacheAdd: () => log.info('Creating new letter service') });
+export const getInteropService = moize(createInteropService, { onCacheAdd: () => log.info('Creating new letter service') });
```
Comment (gregory-j-baker) [inline frontend/app/services/interop-service.server.ts:34]:
```suggestion
export const getInteropService = moize(createInteropService, { onCacheAdd: () => log.info('Creating new interop service') });
```

**Review by gregory-j-baker — DISMISSED** [review]
Sorry.. One tiny change requested.

File: frontend/app/mocks/power-platform-api.server.ts [hunk 2]
```diff
@@ -96,9 +107,37 @@ export function getPowerPlatformApiMockHandlers() {
...
+      return HttpResponse.json(
+        letterTypesEntities?.map((letterType) => {
+          return {
+            nameEn: letterType.nameEn,
+            nameFr: letterType.nameFr,
+            code: letterType.code,
+            id: letterType.id,
+          };
+        }),
+      );
```
Comment (sebastien-comeau) [inline frontend/app/mocks/power-platform-api.server.ts:125]:
```suggestion
      http.get('https://api.example.com/letters-types', () => {
      return HttpResponse.json(
        getAllLetterTypes().map((letterType) => ({
          nameEn: letterType.nameEn,
          nameFr: letterType.nameFr,
          code: letterType.code,
          id: letterType.id,
        })),
      );
    }),
```

File: frontend/app/mocks/power-platform-api.server.ts [hunk 3]
```diff
@@ -96,9 +107,37 @@ export function getPowerPlatformApiMockHandlers() {
...
 
+/**
+ * Retrieves list of letter types.
+ *
+ * @returns All the letter types found.
+ */
+export function getAllLetterTypes() {
+  const letterTypesEntities = db.letterType.getAll();
+
+  return letterTypesEntities;
```
Comment (sebastien-comeau) [inline frontend/app/mocks/power-platform-api.server.ts:138]:
```suggestion
  return db.letterType.getAll();
```

File: frontend/app/routes/_gcweb-app.letters._index.tsx [hunk 4]
```diff
@@ -91,7 +91,7 @@ export default function LettersIndex() {
             <Link reloadDocument to={`/letters/${letter.referenceId}/download`} className="font-medium hover:underline">
               {i18n.language === 'en' ? letter.nameEn : letter.nameFr}
             </Link>
-            <p className="mt-1 text-sm text-gray-500">{t('letters:index.date', { date: getFormattedDate(letter.dateSent) })}</p>
+            <p className="mt-1 text-sm text-gray-500">{t('letters:index.date', { date: getFormattedDate(letter?.dateSent) })}</p>
```
Comment (sebastien-comeau) [inline frontend/app/routes/_gcweb-app.letters._index.tsx:94]:
```suggestion
            <p className="mt-1 text-sm text-gray-500">{t('letters:index.date', { date: getFormattedDate(letter.dateSent) })}</p>
```

File: frontend/app/mocks/power-platform-api.server.ts (hunk 2 above)
Comment (gregory-j-baker) [reply frontend/app/mocks/power-platform-api.server.ts:125]: This should be `letter-types` not `letters-types`.

File: frontend/app/services/interop-service.server.ts (hunk 1 above)
Comment (carolinegeorge-gov) [reply frontend/app/services/interop-service.server.ts]: Fixed: c484a60f267a6549d761d0e41ff16f7040ea61e6

File: frontend/app/routes/_gcweb-app.letters._index.tsx (hunk 4 above)
Comment (carolinegeorge-gov) [reply frontend/app/routes/_gcweb-app.letters._index.tsx]: Fixed: c484a60f267a6549d761d0e41ff16f7040ea61e6

File: frontend/app/mocks/power-platform-api.server.ts (hunk 3 above)
Comment (carolinegeorge-gov) [reply frontend/app/mocks/power-platform-api.server.ts]: Fixed: c484a60f267a6549d761d0e41ff16f7040ea61e6

File: frontend/app/mocks/power-platform-api.server.ts (hunk 2 above)
Comment (carolinegeorge-gov) [reply frontend/app/mocks/power-platform-api.server.ts]: Fixed: c484a60f267a6549d761d0e41ff16f7040ea61e6 and 45ef18f0a25a877da32b0caf302ed7bf2e424159

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**